### PR TITLE
Reading the Balance port (if present)

### PR DESCRIPTION
**Pull request by markmellors**
seems to work well. gives a useful message if there's a problem. Currently the program doesn't change behaviour if there's a problem though.

### CODE REVIEW

**Review by robberwick — CHANGES_REQUESTED** [review]
looks good. I think it's worth extracting the `ADS1x15` library out into it's own thing, and ther are a couple of other very minor comments

File: libs/balance_port/include/balance_port.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
+
+#include <stdio.h>
+#include <string>
+#include "pico/stdlib.h"
+#include "hardware/i2c.h"
+#include "ads1x15.h"
```
[the remark this answers is not in the document]
Comment (markmellors) [reply libs/balance_port/include/balance_port.h]: `ADS1x15` is now a seperate (external) library, but the `balance_port` library is still tightly coupled to it.

File: libs/receiver/src/receiverCPPM.cpp [hunk 2]
```diff
@@ -36,6 +36,6 @@ ReceiverChannelValues ReceiverCPPM::get_channel_values() const {
 }
 
 bool ReceiverCPPM::get_receiver_data() const {
-    return decoder->getFrameAgeMs() > 0;
+    return decoder->getFrameAgeUs() > 0;
```
[the remark this answers is not in the document]
Comment (markmellors) [reply libs/receiver/src/receiverCPPM.cpp:39]: no, but it was needed to get my code to compile. This change is now on `main`. I'd cherrypicked the commit from  #48 , so expected this change to disappear with a rebase. I've now rebased, but it's made it worse, we now appear to have two commits making this change! I'm confused.

**Review by robberwick — APPROVED** [review]
Yes :+1:

**Pull request merged**